libs/agno/agno/models/base.py

```diff
@@ -417,10 +417,21 @@ def response(
                             ]
                             and function_call_response.tool_executions is not None
                         ):
+                            # Record the tool execution in the model response
                             if model_response.tool_executions is None:
                                 model_response.tool_executions = []
                             model_response.tool_executions.extend(function_call_response.tool_executions)
 
+                            # If the tool is currently paused (HITL flow), add the requirement to the run response
+                            if (
+                                function_call_response.event == ModelResponseEvent.tool_call_paused.value
+                                and run_response is not None
+                            ):
+                                current_tool_execution = function_call_response.tool_executions[-1]
```

Comment:
Could there be multiple tool_executions?

Reply:
I will double check